src/GitHub.Exports/Helpers/ThreadingHelper.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Threading.Tasks;
+using ThreadHelper = Microsoft.VisualStudio.Shell.ThreadHelper;
+using GitHub.Extensions;
+
+namespace GitHub.Helpers
+{
+    public static class ThreadingHelper
+    {
+        public async static Task SwitchToMainThreadAsync()
+        {
+            if (!Guard.InUnitTestRunner())
```

Comment:
😁.. only pointed it out 'cause I tried doing that exact thing the other week and got confused when my unit tests continued to fail. That was before I realised that, of course, it wouldn't work (when you get your head around how async/await works, in terms of threading).